tools/tcat_ble_client/cli/base_commands.py

```diff
@@ -54,55 +55,112 @@ async def execute_default(self, args, context):
         return CommandResultNone()
 
 
-class HelloCommand(Command):
+class BleCommand(Command):
 
-    def get_help_string(self) -> str:
-        return 'Send round trip "Hello world!" message.'
+    @abstractmethod
+    def get_log_string(self) -> str:
+        pass
+
+    @abstractmethod
+    def prepare_data(self, context):
+        pass
 
     async def execute_default(self, args, context):
         bless: BleStreamSecure = context['ble_sstream']
-        print('Sending hello world...')
-        data = TLV(TcatTLVType.APPLICATION.value, bytes('Hello world!', 'ascii')).to_bytes()
+
+        print(self.get_log_string())
+        data = self.prepare_data(context)
         response = await bless.send_with_resp(data)
         if not response:
             return
         tlv_response = TLV.from_bytes(response)
         return CommandResultTLV(tlv_response)
 
 
-class CommissionCommand(Command):
+class HelloCommand(BleCommand):
+
+    def get_log_string(self) -> str:
+        return 'Sending hello world...'
+
+    def prepare_data(self, context):
+        return TLV(TcatTLVType.APPLICATION.value, bytes('Hello world!', 'ascii')).to_bytes()
 
     def get_help_string(self) -> str:
-        return 'Update the connected device with current dataset.'
+        return 'Send round trip "Hello world!" message.'
 
-    async def execute_default(self, args, context):
-        bless: BleStreamSecure = context['ble_sstream']
-        dataset: ThreadDataset = context['dataset']
 
-        print('Commissioning...')
+class CommissionCommand(BleCommand):
```

Comment:
nit: can we keep the same order of methods implementation in each class to improve readability? `HelloCommand` and `CommissionCommand` have a different order than all other classes

Reply:
done